homeassistant/loader.py

```diff
@@ -940,22 +979,28 @@ def get_component(self) -> ComponentProtocol:
             )
             raise ImportError(f"Exception importing {self.pkg_path}") from err
 
-        if self.platform_exists("config"):
-            # Setting up a component always checks if the config
-            # platform exists. Since we may be running in the executor
-            # we will use this opportunity to cache the config platform
-            # as well.
-            with suppress(ImportError):
-                self.get_platform("config")
+        if self.integration_type not in SKIP_PRELOAD_INTEGRATION_TYPES:
+            for platform_name in self._preload_platforms:
+                if self.platform_exists(platform_name):
```

Comment:
This isn't great to do in a loop. Maybe platforms exist

Reply:
I think we can make that work now

Input is a list, output is a list that exist

Raises if top level comp is not loaded now because we shouldn't ever call it its it's not